projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionLogsUrlBuilder.java

```diff
@@ -101,7 +104,8 @@ private String convertDate(OffsetDateTime dateTime, Long offset) {
       String format = dateFormat;
 
       if (dateTime == null) {
-         return "";
+         // if not set it to current time
+         dateTime = OffsetDateTime.now(clock);
```

Comment:
OffsetDateTime.now() uses the system default zone (Clock.systemDefaultZone()) by default.

Reply:
Yes. But I want to be able to mock the Clock to a fixed clock in teh tests. That's why I am passing it explicitly.